src/main/java/com/epam/aidial/core/service/LockService.java

```diff
@@ -42,6 +46,21 @@ public Lock lock(String key) {
         return () -> unlock(id, owner);
     }
 
+    public <T> T underUserLock(Proxy proxy, ProxyContext context, Supplier<T> function) {
+        String userBucketLocation = BlobStorageUtil.buildInitiatorBucket(context);
+        String prefix = proxy.getStorage().getPrefix();
```

Comment:
The logic to handle the prefix case could be moved to BlobStorageUtil. Someone could forget add this logic.